src/config/settings.ts

```diff
@@ -118,6 +118,102 @@ export async function clear() {
   await fs.promises.rm(paths.config, { recursive: true, force: true } as any);
 }
 
+function getConfigLHS(cfg: any, fqFieldName: string): [any, string] | null {
+  const optionParts = fqFieldName.split('-');
+  const finalOptionPart = optionParts.pop() ?? '';
+  const lhs = optionParts.reduce((currentConfig, currentField) => {
+    // Allow for case-insensitive searching if the field name is all lower-case
+    if (!currentConfig) {
+      return null;
+    } else if (currentField in currentConfig) {
+      return currentConfig[currentField];
+    } else {
+      return null;
+    }
+  }, cfg);
+
+  if (!lhs) {
+    return null;
+  } else if (finalOptionPart in lhs) {
+    return [lhs, finalOptionPart];
+  } else {
+    return null;
+  }
+}
+
+export function updateFromCommandLineAndSave(cfg: Settings, args: string[]): Settings {
+  const newConfig = updateFromCommandLine(cfg, args);
+
+  save(newConfig);
+
+  return newConfig;
+}
+
+// For testing purposes only
```

Comment:
This function exists only for testing purposes? (You may need to clarify the comment.)
Ideally, also document the function…

Reply:
Also, consider mocking `paths.settings` in the test to drop the need to have a separate function (and bonus: also get to test the save functionality).

Reply:
The most straightforward way was to mock `fs.writeFileSync` because mocking typescript modules is harder -- and after spending some time, I still don't know how to do that.